x-pack/plugins/enterprise_search/public/applications/app_search/components/engine/utils.ts

```diff
@@ -6,12 +6,14 @@
 
 import { generatePath } from 'react-router-dom';
 
+import { encodePathParams } from '../../utils/encode_path_params';
+
 import { EngineLogic } from './';
 
 /**
  * Generate a path with engineName automatically filled from EngineLogic state
  */
 export const generateEnginePath = (path: string, pathParams: object = {}) => {
   const { engineName } = EngineLogic.values;
-  return generatePath(path, { engineName, ...pathParams });
+  return generatePath(path, encodePathParams({ engineName, ...pathParams }));
```

Comment:
My hope is that by baking this into our new `generateEnginePath` helper that it'll "just work" going forward and at the very least we won't accidentally forget to encode user-generated IDs going forward.

Reply:
It'll just work if someone is using this helper... but aren't there places in our code that we are generating engine paths without this helper?

Reply:
I think the problem I see here is that you are actually increasing the odds that someone will accidentally forget to encode user-generate ids, because as soon as you're in a place in the code you're not able to use `generateEnginePath`, then you won't be in the habit of encoding path params.

I really wish there was just one way to create paths that we could use *anywhere*, decoupled from state, that would handle encoding path params.

```
generateAppPath(ENGINE_DOCUMENT_DETAIL_PATH, {
  engineName: resultMeta.engine,
  documentId: resultMeta.id,
})
```